### PR TITLE
don't check dataset labels during preprocess for GRPO

### PR DESCRIPTION
**Pull request by winglian**
# Description

## Summary by CodeRabbit

* **Bug Fixes**
  * Restricted debug label checking for datasets to cases where the RL type is not "ORPO," reducing unnecessary log output in specific configurations.

### CODE REVIEW

File: src/axolotl/common/datasets.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def load_preference_datasets(
             math.ceil(len(train_dataset) * cfg.num_epochs / cfg.batch_size)
         )
 
-    if (cli_args and cli_args.debug) or cfg.debug:
+    if ((cli_args and cli_args.debug) or cfg.debug) and cfg.rl != "grpo":
```
[the remark this answers is not in the document]
Comment (NanoCode012) [reply src/axolotl/common/datasets.py]: I think this suggestion is better. For now, we can exclude grpo, but others rl may need this too.

[the remark this answers is not in the document]
Comment (salmanmohammadi) [reply src/axolotl/common/datasets.py]: @winglian also agree with this suggestion. Otherwise LGTM.

**Review by salmanmohammadi — APPROVED** [review]
